### PR TITLE
Add image suggestion feature to the iMatrics service

### PR DESCRIPTION
**Pull request by imatricsOpenSource**
Hi! These changes adds support for an image suggestion widget in the fronted that suggests images based on the tags on the article.

Two new environment variables are required. We can discuss them in our shared slack channel.

Happy reading!

### CODE REVIEW

File: superdesk/text_checkers/ai/__init__.py [hunk 1]
```diff
@@ -156,6 +158,70 @@ def create(self, docs, **kwargs):
         return [0]
 
 
+class AIImageResource(Resource):
+    schema = {
+        "service": {
+            "type": "string",
+            "required": True,
+        },
+        "item": {
```
Comment (petrjasek) [inline superdesk/text_checkers/ai/__init__.py:167]: would just call this `items` if it's a list and not a single one

Comment (petrjasek) [reply superdesk/text_checkers/ai/__init__.py:167]: or make it `item` which won't be a list but a dictionary like in the other services

Comment (hanstchou) [reply superdesk/text_checkers/ai/__init__.py]: Thank you for the feedback. We have now adjusted this and refactored from "item" to "items".

File: superdesk/text_checkers/ai/imatrics.py [hunk 2]
```diff
@@ -208,6 +216,27 @@ def _analyze(self, data, **params):
...
 
+    def search_images(self, items: list) -> dict:
+        """fetch image suggestions"""
+        if not self.base_url or not self.user or not self.key:
+            logger.warning("IMatrics is not configured properly, can't fetch images")
+            return {"result": []}
+        data = items
+        try:
+            r_data = self._search_images(data)
+        except:
```
Comment (petrjasek) [inline superdesk/text_checkers/ai/imatrics.py:227]: would be useful to log the exception

**Review by petrjasek — APPROVED** [review]

File: superdesk/text_checkers/ai/imatrics.py (hunk 2 above)
Comment (hanstchou) [reply superdesk/text_checkers/ai/imatrics.py:227]: We were told to add filtering of images without src in the server instead, therefore we made a small change after the approval.

**Pull request closed without merging**